File: test/annotations/cravat_summary.py
#!/usr/bin/env python
import csv
from xlsxwriter.workbook import Workbook

tsv_file = 'summary/tsv/cravat_summary.tsv'
xlsx_file = 'summary/cravat_summary.xlsx'

workbook = Workbook(xlsx_file)
worksheet = workbook.add_worksheet()

tsv_reader = csv.reader(open(tsv_file, 'rb'), delimiter='\t')

for row, data in enumerate(tsv_reader):
    worksheet.write_row(row, 0, data)

workbook.close()

File: test/annotations/cravat_summary.R
#!/usr/bin/env Rscript

suppressPackageStartupMessages(library("optparse"))

if (!interactive()) {
    options(warn = -1, error = quote({ traceback(); q('no', status = 1) }))
}

args_list = list(make_option("--sample_names", default = NA, type = 'character', help = "sample name"))
				  
parser = OptionParser(usage = "%prog", option_list = args_list)
arguments = parse_args(parser, positional_arguments = T)
opt = arguments$options

sample_names = unlist(strsplit(x=opt$sample_names, split=" ", fixed=TRUE))
tsv = list()
for (i in 1:length(sample_names)) {
	tsv[[i]] = read.csv(file=paste0("cravat/", sample_names[i], ".txt"), header=TRUE, sep="\t", stringsAsFactors=FALSE)
}
tsv = do.call(rbind, tsv)
write.table(tsv, file="summary/tsv/cravat_summary.tsv", sep="\t", col.names=TRUE, row.names=FALSE, quote=FALSE, append=FALSE)
